src/driver/clint.rs

```diff
@@ -18,32 +14,32 @@ pub const MSIP_WIDTH: Width = Width::Byte4;
 pub const MTIMECMP_WIDTH: Width = Width::Byte8;
 pub const _MTIME_WIDTH: Width = Width::Byte8;
 
-#[derive(Clone, Debug)]
-pub struct ClintDriver {
-    /// The base address of the physical CLINT.
-    base: usize,
-}
+pub mod clint_driver {
```

Comment:
We do not want to have a single global driver. In general, any kind of "ambient authority" is often bad design and ends up biting back in the long term.
Here a typical issue would be supporting NUMA nodes with two CLINTs.
Another issue would be determining dynamically the driver we should use (e.g. if we want to support ACLINT features, as we have one the P550).

So in short, the previous architecture is already functional, the only modifications needed is to remove the `Mutex`and pass `&self` instead of `&mut self` as arguments.